core/src/services/routing.js

```diff
@@ -299,6 +299,47 @@ export const handleRouteChange = async (path, component, node, config) => {
     return;
   }
   try {
+    if (
+      component &&
+      typeof component.get === 'function' &&
+      component.get().unsavedChanges &&
+      component.get().unsavedChanges.isDirty
+    ) {
+      const newUrl = window.location.href;
+      const oldUrl = component.get().unsavedChanges.persistUrl;
+
+      //pretend the url hasn't been changed
+      oldUrl && history.replaceState(window.state, '', oldUrl);
+
+      if (
+        typeof component.showModal !== 'function' ||
+        typeof component.hideModal !== 'function'
+      ) {
+        return; //in case provided component doesn't have this functionality
+      }
+      component
```

Comment:
Here `component` is always 'AppComponent'. Therefore the inputs to showModal method could be removed and rewrite the showModal function as it follows:
```
      showModal() {
        const title = 'Unsaved changes detected';
        const text = 'It looks like you might lose some data if you leave this page. Are you sure you want to do this?'
...
```
This does not affect the reusability of the Confirmation modal component

Reply:
It does, you are unable to use the confirmation modal for any other purpose than this one.

Reply:
I see your point, it does not affect reusing the ConfirmationModal component in other component, but it does for reusing it on App component